sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOIT.java

```diff
@@ -0,0 +1,168 @@
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+
+/**
+ * IOIT test to run HCatalog.
+ * <p>You can run tests on prepared hCatalog infrastructure.</p>
+ * <p>To run test specify number of records (numberOfRecords) to write to HCatalog,
+ * metastore url (HCatalogMetastoreHostName),
+ * metastore port (HCatalogMetastorePort),
+ * hive port (HCatalogHivePort)
+ * and hive database (HCatalogHiveDatabase)
+ * to create a hive test table.</p>
+ * <pre>{@code mvn clean verify -Pio-it -pl sdks/java/io/hcatalog/
+ * -DintegrationTestPipelineOptions=
+ * '[
+ * "--tempRoot=gs://url",
+ * "--runner=TestDataflowRunner",
+ * "--numberOfRecords=100",
+ * "--HCatalogMetastoreHostName=hcatalog-metastore",
+ * "--HCatalogMetastorePort=9083"
+ * "--HCatalogHivePort=10000"
+ * "--HCatalogHiveDatabaseName=default"
+ * ]'
+ * }</pre>
+ */
+
+@RunWith(JUnit4.class)
+public class HCatalogIOIT {
+
+  private static final Map<Integer, String> EXPECTED_HASHES = ImmutableMap.of(
+      100, "34c19971bd34cc1ed6218b84d0db3018",
+      1000, "2db7f961724848ffcea299075c166ae8",
+      10_000, "7885cdda3ed927e17f7db330adcbebcc"
+  );
+
+  private static String tableName;
+  private static HiveDatabaseTestHelper helper;
+  private static Map<String, String> configProperties;
+  private static Integer numberOfRecords;
+  private static String metastoreUri;
+  private static Integer metastorePort;
+  private static Integer hivePort;
+  private static String hostName;
+  private static String databaseName;
+
+  @Rule
+  public TestPipeline pipelineWrite = TestPipeline.create();
+  @Rule
+  public TestPipeline pipelineRead = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    IOTestPipelineOptions options = TestPipeline.testingPipelineOptions()
+        .as(IOTestPipelineOptions.class);
+
+    numberOfRecords = options.getNumberOfRecords();
+    hostName = options.getHCatalogMetastoreHostName();
+    metastorePort = options.getHCatalogMetastorePort();
+    metastoreUri = String.format("thrift://%s:%s", hostName, metastorePort);
+
+    configProperties = ImmutableMap.of("hive.metastore.uris", metastoreUri);
+
+    hivePort = options.getHCatalogHivePort();
+    databaseName = options.getHCatalogHiveDatabaseName();
+    helper = new HiveDatabaseTestHelper(
+        hostName,
+        hivePort,
+        databaseName);
+
+    try {
+      tableName = helper.createHiveTable("HCatalogIOIT");
+    } catch (Exception e) {
+      helper.closeConnection();
+      throw new Exception("Problem with creating table. " + e);
+    }
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    try {
+      helper.dropHiveTable(tableName);
+    } catch (Exception e) {
+      helper.closeConnection();
+      throw new Exception("Problem with deleting database.");
```

Comment:
Maybe better `"Problem deleting table " + table + ": " + e` for consistency. don't forget to propragate the original Exception too `Exception("Problem....", e);`

Reply:
OK.